lib/facter/custom_facts/util/fact.rb

```diff
@@ -173,7 +173,7 @@ def find_suitable_resolutions(resolutions)
       end
 
       def sort_by_weight(resolutions)
-        resolutions.sort { |a, b| b <=> a }
+        resolutions.sort_by.with_index { |x, idx| [-x.weight, -(x.fact_type.to_s.bytes.inject(:+) || 0), idx] }
```

Comment:
This is just a test to overcome Ruby's broken sort mechanism.
https://bugs.ruby-lang.org/issues/1089